kyverno-test-util.py

```diff
@@ -0,0 +1,132 @@
+import sys
+import yaml
+import os
+from types import NoneType
+
+# ReplicaSet is not here, because I have no idea how to change policies to make it work
+allowedResources = ['Job', 'Deployment', 'StatefulSet']
+ignoredFileNames = ['prometheus-scrapeConfig', 'chart', 'Chart',
+                    'values', 'CustomResourceDefinition', 'crds']
+
+kyvernoTestFileName = 'kyverno-test.yaml'
+
+def deleteLeftovers():
+    if os.path.exists(kyvernoTestFileName):
+        os.remove(kyvernoTestFileName)
+
+def writeTitle():
+    title = {'name': 'Kyverno test'}
+    appendToFile(title)
+
+def writePolicies(policies_location):
+    policies = {'policies': [f'{policies_location}']}
+    appendToFile(policies)
+
+def getResourceLocations(folder):
+    """
+    Get's a list of resources that should be validated against policies.
+    """
+    resources = list()
+
+    for root, dirs, files in os.walk(f'{folder}'):
+        for file in files:
+            if (file.endswith(".yaml") or file.endswith(".yml")) and ("kyverno" not in root) and len([ignored for ignored in ignoredFileNames if (file.find(ignored) != -1)]) == 0:
+                resources.append(os.path.join(root, file))
+    return resources
+
+def writeResources(folder):
+    resources = {'resources': getResourceLocations(folder)}
+    appendToFile(resources)
+
+def writeResults(folder, policies_location):
+    results = {'results': getPreparedResults(folder, policies_location)}
+    appendToFile(results)
+
+def getExpectedResult(rule, resource):
+    """
+    This method is meant to determine what should be the result of given test.
+    We don't have tests that are meant to fail, only pass, or if resource is to be ignored - skip.
+    """
+    shouldExclude = 0
+    if 'exclude' in rule.keys():
+        if 'namespace' in str(rule['exclude']) and resource['metadata']['namespace'] in rule['exclude']['any'][0]['resources']['namespaces']:
+            shouldExclude = 1
+
+        if 'kinds' in str(rule['exclude']) and resource['metadata']['name'] in rule['exclude']['any'][0]['resources']['names']:
+            shouldExclude = 1
+
```

Comment:
mutating tests always fail. For the moment, I added the following. Maybe someone knows a better solution?
```suggestion
    if 'mutate' in rule.keys():
        shouldExclude = 1
        
```

Reply:
Oh, that's most likely because I don't have any in my setup, and this script never accounted for those.